code/modules/atmospherics/machinery/components/trinary_devices/filter.dm

```diff
@@ -43,27 +43,19 @@
 	..()
 
 /obj/machinery/atmospherics/components/trinary/filter/update_icon_nopipes()
-
-	if(!(stat & NOPOWER) && on && NODE1 && NODE2 && NODE3)
+	if(is_operational() && on && NODE1 && NODE2 && NODE3)
 		icon_state = "filter_on[flipped?"_f":""]"
 		return
-
 	icon_state = "filter_off[flipped?"_f":""]"
 
 /obj/machinery/atmospherics/components/trinary/filter/power_change()
-	var/old_stat = stat
 	..()
-	if(stat & NOPOWER)
-		on = FALSE
-	if(old_stat != stat)
-		update_icon()
+	update_icon()
```

Comment:
Why did you remove the unchanged check ? power change is not always affecting every machine. Eg. light channel change won't affect atmos machinery but will still update them.

Reply:
The other components don't have these checks so I figured I should make filters act the same.

You think having all of them check for unchanged would be better?

Reply:
In this case it was there because update_icon did some icon operations which can get expensive, but probably wouldn't hurt in general.